cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/meta/RemoteLineageWriterHandler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers.meta;
+
+import co.cask.cdap.data2.metadata.lineage.AccessType;
+import co.cask.cdap.data2.metadata.writer.LineageWriter;
+import co.cask.cdap.internal.app.store.remote.MethodArgument;
+import co.cask.cdap.proto.Id;
+import co.cask.http.HttpResponder;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+
+import java.util.Iterator;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+
+/**
+ * The {@link co.cask.http.HttpHandler} for handling REST calls to LineageStore.
+ */
+@Path(AbstractRemoteStoreHandler.VERSION + "/execute")
```

Comment:
I suggested a plural noun to indicate a 'contains' relationship with the individual operations because that's more RESTful than using the verb 'execute'